todo_app/app.py

```diff
@@ -21,10 +52,19 @@ def index():
         else:
             items.sort(key=lambda x: x.due_date)
 
-        return render_template('index.html', view_model=item_view_model, todays_date=todays_date)
+
+        if hasattr(current_user, 'role'):
+            if current_user.role == 'writer':
+                return render_template('index.html', view_model=item_view_model, todays_date=todays_date)   
+            else:
+                return render_template('index_ro.html', view_model=item_view_model, todays_date=todays_date)
+        else:
+            return render_template('index.html', view_model=item_view_model, todays_date=todays_date)
+
 
 
     @app.route('/new_item', methods=['POST'])
+    @login_required
     def new_item():
         new_item_title = request.form.get('new_item_title')
```

Comment:
You need to check the user is a writer here! Don't rely on the fact that you aren't presenting them with the button on the page - anyone could craft an HTTP request that hits this endpoint.